File: mandel.py
def mandel(x, y, max_iters, value):
    """
    Given the real and imaginary parts of a complex number,
    determine if it is a candidate for membership in the Mandelbrot
    set given a fixed number of iterations.
    """
    i = 0
    c = complex(x,y)
    z = 0.0j
    for i in range(max_iters):
        z = z*z + c
        if (z.real*z.real + z.imag*z.imag) >= 4:
            value[0] = i
            return 0
    value[i] = 255
    return 1

File: mandel.c
int mandel(float x, float y, int max_iters, unsigned char * val)
{
    int i = 0;
    float cR = x;
    float cI = y;
    float zR = 0;
    float zI = 0;
    for (i = 0; i < max_iters; i++)
    {
        /* in complex notation, z * z + c */
        zR = zR * zR - zI * zI + cR;
        zI = 2 * zR * zI + cI;
        if ((zR * zR + zI * zI) >= 4)
        {
            *val = i;
            return 0;
        }
    }
    *val = max_iters;
    return 1;
}
